src/js/controller/settings/exportimage/PngExportController.js

```diff
@@ -156,6 +158,52 @@
     });
   };
 
+  ns.PngExportController.prototype.onPixiDownloadClick_ = function () {
+    var zip = new window.JSZip();
+
+    // Create PNG export.
+    var canvas = this.createPngSpritesheet_();
+    var name = this.piskelController.getPiskel().getDescriptor().name;
+
+    zip.file(name + '.png', pskl.utils.CanvasUtils.getBase64FromCanvas(canvas) + '\n', {base64: true});
+
+    var width = canvas.width / this.getColumns_();
+    var height = canvas.height / this.getRows_();
+
+    var numFrames = this.piskelController.getFrameCount();
+    var frames = {};
+    for (var i = 0; i < numFrames; i++) {
+      var column = i % this.getColumns_();
+      var row = (i - column) / this.getColumns_();
+      var frame = {
+        'frame': {'x': width * column,'y': height * row,'w': width,'h': height},
+        'rotated': false,
+        'trimmed': false,
+        'spriteSourceSize': {'x': 0,'y': 0,'w': width,'h': height},
+        'sourceSize': {'w': width,'h': height}
+      };
+      frames[name + i + '.png'] = frame;
```

Comment:
Yes, looks like this was really specific to this example. I think your implementation is fine.